cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -444,8 +585,32 @@ async function domainResources (alias) {
             hostedZoneID: await hostedZoneID.root(),
         };
     }
-    throw new Error(`unrecognized domain type for ${alias}`);
+    throw new UnrecognizedDomainTypeError(`unrecognized domain type for ${alias}`);
+}
+
+/**
+ * Returns true of setA is equal to setB.
+ * @param setA
+ * @param setB
+ * @returns {boolean}
+ */
+function setEqual(setA, setB) {
+    if (setA.size !== setB.size) {
+        return false;
+    }
+
+    for (let elem of setA) {
+        if (!setB.has(elem)) {
+            return false;
+        }
+    }
```

Comment:
nit we could make it more efficient by sorting the elements and then join and compare the string, or compare them one by one.

Reply:
nvm it's a set so the current implementation is faster